### PR TITLE
Add attribute 'syslog', avoid the syslog configuration for certain node

### PR DESCRIPTION
**Pull request by xuweibj**
Story 86265
If set 'syslog' as ignore, the syslog script will not run in postscript.

### CODE REVIEW

File: xCAT/postscripts/syslog [hunk 1]
```diff
@@ -13,6 +13,10 @@
 #           OSVER: possible values are sles10, fedora8, redhat5, aix etc.   
 #=cut
 #-------------------------------------------------------------------------------
+if [ "$SYSLOG" = "ignore" ]; then
+    exit 0
+fi
```
Comment (zet809) [inline xCAT/postscripts/syslog]: If the value is not "ignore", will still run this script?

File: perl-xCAT/xCAT/Schema.pm [hunk 2]
```diff
@@ -668,6 +668,7 @@ noderes => {
   routenames => 'A comma separated list of route names that refer to rows in the routes table. These are the routes that should be defined on this node when it is deployed.',
   nameservers => 'An optional node/group specific override for name server list.  Most people want to stick to site or network defined nameserver configuration.',
   proxydhcp => 'To specify whether the node supports proxydhcp protocol. Valid values: yes or 1, no or 0. Default value is yes.',
+  syslog => "Set to 'ignore', will not run syslog script in postscript",
```
Comment (zet809) [inline perl-xCAT/xCAT/Schema.pm]: I think you can add more comments here, such as it is used to configure syslog for the compute node, currently, only 'ignore' is supported.

File: perl-xCAT/xCAT/Schema.pm [hunk 3]
```diff
@@ -1846,6 +1847,10 @@ my @nodeattrs = (
                  tabentry => 'noderes.proxydhcp',
                  access_tabentry => 'noderes.node=attr:node',
   },
+        {attr_name => 'syslog',
```
Comment (zet809) [inline perl-xCAT/xCAT/Schema.pm]: Pls keep the format to be consistent.

File: perl-xCAT/xCAT/Schema.pm [hunk 4]
```diff
@@ -668,6 +668,7 @@ noderes => {
   routenames => 'A comma separated list of route names that refer to rows in the routes table. These are the routes that should be defined on this node when it is deployed.',
   nameservers => 'An optional node/group specific override for name server list.  Most people want to stick to site or network defined nameserver configuration.',
   proxydhcp => 'To specify whether the node supports proxydhcp protocol. Valid values: yes or 1, no or 0. Default value is yes.',
+  syslog => "It is used to configure syslog for the compute node. Currently, only 'ignore' is supported, which means will not run syslog script in postscript",
```
Comment (daniceexi) [inline perl-xCAT/xCAT/Schema.pm]: I suggest the message to be:
~~~~
To configure how to configure syslog for compute node. Valid values:blank, ignore. blank - run postscript syslog.; ignore - Do NOT run postscript syslog;
~~~~

**Comment by zet809** [comment]
Looks good to me, I agree to merge.

**Comment by whowutwut** [comment]
@xuweibj  In the description of this pull request, can you add the external usage for this new option.  
For users of xCAT, how would they set it? 

I also think we should have FVT create a testcase for this to verify that syslog is executed or not executed.

**Comment by whowutwut** [comment]
Another comment I have... with the changes to Schema.pm, why is there so much differences? Did you run perl tidy?  If we are doing any type of tidy code, we should put that first into it's own commit, then have code changes in another commit.  There's no way to review the changes when git cannot display the diffs .   @daniceexi  FYI

**Comment by whowutwut** [comment]
@zet809  looking at your comments before and my email notifications, it looks like the commit was done using the --amend and caused over 3,000 changes to Schema.pm.. what happened here? 

Is this really what we should be doing once we start a code review? It's very difficult for us to see the changes made from the review comments...

**Comment by daniceexi** [comment]
@xuweibj @zet809 @whowutwut I totally agree, for the change of tidying a file, it must have an individual commit so that we can just review it quickly to make sure there's no format issue.

**Comment by whowutwut** [comment]
with this changes in Schema.pm, this resulted in changes to the .rst files when running ``create_man_pages.py``..  I think we need to build the rst file and look at the resulting output. 
```
[vhu@victorvm7 xcat-core]$ git status
# On branch master
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#	modified:   docs/source/guides/admin-guides/references/man5/mpa.5.rst
#	modified:   docs/source/guides/admin-guides/references/man5/noderes.5.rst
#	modified:   docs/source/guides/admin-guides/references/man7/group.7.rst
#	modified:   docs/source/guides/admin-guides/references/man7/node.7.rst
```

I'm not sure if the changes are correct...   I quickly looked at a diff for one page..  mpa.rst.
```
[vhu@victorvm7 xcat-core]$ git diff docs/source/guides/admin-guides/references/man5/mpa.5.rst
diff --git a/docs/source/guides/admin-guides/references/man5/mpa.5.rst b/docs/source/guides/admin-guides/references/man5/mpa.5.rst
index f06f0bb..e44e0f8 100644
--- a/docs/source/guides/admin-guides/references/man5/mpa.5.rst
+++ b/docs/source/guides/admin-guides/references/man5/mpa.5.rst
@@ -63,9 +63,9 @@ mpa Attributes:
 \ **slots**\ 
  
  The number of available slots in the chassis. For PCM, this attribute is used to store the number of slots in the following format:  <slot rows>,<slot columns>,<slot orient
-    <slot rows>  = number of rows of slots in chassis
-    <slot columns> = number of columns of slots in chassis
-    <slot orientation> = set to 0 if slots are vertical, and set to 1 if slots of horizontal
+                  <slot rows>  = number of rows of slots in chassis
+                  <slot columns> = number of columns of slots in chassis
+                  <slot orientation> = set to 0 if slots are vertical, and set to 1 if slots of horizontal
  
```
I didn't build the html man pages but that doesn't look like it would render right..    maybe it will..  @xuweibj  can you double check?

**Comment by daniceexi** [comment]
@whowutwut @xuweibj made the perltidy against the Schema.pm so that there were some indent changes in the description string. Then it caused the change in the map page.

**Pull request merged**